### ISSUE TITLE
No ping-status 'OK' when script has no stderr/out?

### ISSUE THREAD

**Issue by lfkdev** [bdd/runitor#29]
Hi, great tool! But i found the following problem.
When there is no output from the script, runitor seems to think it's still running.
Script to reproduce it:
```
#!/usr/bin/env bash
  
foo() {
  bar=$(ls /)
}
foo
```
cron entry:
```
HC_API_URL="https://mydomain.com/ping/"
* * * * * /usr/bin/runitor --uuid="xxxx-xxx-x-x-xxxx" -- bash /usr/local/bin/test.sh
```

The Script works without any problems, but my selfhosted healthchecks.io only shows the "Started" Ping, without the "OK" and the timestamp.
Like here:
```
#46 | Nov 20 | 05:55 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
#45 | Nov 20 | 05:54 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
#44 | Nov 20 | 05:53 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
#43 | Nov 20 | 05:52 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
```

When you now change the script to:
```
#!/usr/bin/env bash
  
foo() {
  bar=$(ls /)
  echo "123"
}

foo
```

It works like it should:
```
#52 | Nov 20 | 05:58 | OK | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor) 0.04 sec
#51 | Nov 20 | 05:58 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
#50 | Nov 20 | 05:57 | OK | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor) 0.03 sec
#49 | Nov 20 | 05:57 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
#48 | Nov 20 | 05:56 | OK | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor) 0.03 sec
#47 | Nov 20 | 05:56 | Started | HTTP POST from 1.2.4.5 - runitor/v0.9.0 (+https://bdd.fi/x/runitor)
```
And shows the output "123" in the body. 

Is this a desired behavior? How can I tell runitor that the script has no output but has still finished?
Thanks!

```
Operating System: Ubuntu 20.04.3 LTS
Kernel: Linux 5.4.0-84-generic
runitor/v0.9.0
```

**Comment by bdd** [comment]
I can't reproduce this with the healthchecks.io instance.

_(Although not relevant to this, that PingStatus logging should not be in the output. It's just noise but I clearly made a `git add -p` mistake 🤦. Will publish 0.9.1 later today.)_

```
~% cat noout.sh
#!/usr/bin/env bash
  
foo() {
  bar=$(ls /)
}
foo

~% runitor -- ./noout.sh
2021/11/20 07:45:45 PingStatus(handle=<redacted> exitCode=0, body)
```
![image](https://user-images.githubusercontent.com/11135/142732750-7abe3141-a505-4864-9ad6-8127466f0053.png)

Can you verify from the HTTP server logs of your own instance, that in fact runitor does not make an HTTP request?

As explained in the [release notes of v0.9.0](https://github.com/bdd/runitor/releases/tag/v0.9.0), exit status is now always delivered. [This commit](https://github.com/bdd/runitor/commit/3bbcb9ae0ec5c4170e6c622bc72a672e20686d61) slightly changed the URI paths hit by runitor.

**v0.8.0**:
start: `<instance-url>/<uuid>/start`
exit code == 0: `<instance url>/<uuid>`
exit code != 0: `<instance_url>/<uuid>/fail`

**v0.9.0**:
start: `<instance-url>/<uuid>/start`
exit: `<instance_url>/<uuid>/<exit-code>`

If the exit code == 0, [Healthchecks interprets it as a successful run](https://healthchecks.io/docs/http_api/#exitcode-uuid).

This feature was added to Healthchecks with in [v.1.18.0](https://github.com/healthchecks/healthchecks/releases/tag/v1.18.0), released Oct 2020, [with this commit](https://github.com/healthchecks/healthchecks/commit/81e59ac553278e700c25ae9d08aa08af1a9b3910).

Is your local instance version older or newer than 1.18.0?

**Comment by lfkdev** [comment]
My HC Version: v1.24.1

In both cases I can see requests in my apache log, here is the log entry with a script that has no output:
```
23.88.53.192 - - [20/Nov/2021:23:00:01 +0000] "POST /ping//xxxx-xxx-xxxx-xxxx/start HTTP/1.1" 200 5767 "-" "runitor/v0.9.0 (+https://bdd.fi/x/runitor)"
23.88.53.192 - - [20/Nov/2021:23:00:01 +0000] "POST /ping//xxxx-xxx-xxxx-xxxx/0 HTTP/1.1" 411 1090 "-" "runitor/v0.9.0 (+https://bdd.fi/x/runitor)"
```

and when it has output:
```
1.2.3.4 - - [20/Nov/2021:23:12:01 +0000] "POST /ping//xxxx-xxx-xxxx-xxxx/start HTTP/1.1" 200 5767 "-" "runitor/v0.9.0 (+https://bdd.fi/x/runitor)"
1.2.3.4 - - [20/Nov/2021:23:12:01 +0000] "POST /ping//xxxx-xxx-xxxx-xxxx/0 HTTP/1.1" 200 1125 "-" "runitor/v0.9.0 (+https://bdd.fi/x/runitor)"
```
The status code differs, intereseting! 411 means "server refuses to accept the request without a defined Content-Length header", do you have an idea what could cause this?

![image](https://user-images.githubusercontent.com/49819192/142743733-92778ff8-07ab-49ad-bfae-88e52dbedd5d.png)

**Comment by bdd** [comment]
Looks like this is similar to #16. Django's development server did not support chunked encoding (header `transfer-encoding: chunked` and no `content-length` header). ...But you mention Apache, which should be implementing the standards rather properly. So that's a bit surprising.

When we _thought_ we fixed chunked encoding in #16, we missed the fact that setting `content-type: 0` is not necessarily enough for empty request bodies. Turns out [since Go 1.8](https://golang.org/doc/go1.8#net_http), the ambiguity of default zero value or explicitly set zero of content-length on HTTP request needs to be handled with `http.NoBody` or `nil` in the server on the other side absolutely cannot handle chunked encoding and wants `content-length: 0`.

Luckily this is essentially a single line addition. https://github.com/bdd/runitor/commit/047cb5fe48d8850e23d49e680ebbd2ab2226733f

I released [v0.9.2](https://github.com/bdd/runitor/releases/tag/v0.9.2) after verifying `content-length: 0` header for empty bodies in status pings.

Can you try this release and report back?

**Comment by lfkdev** [comment]
It works now!
```
1.2.3.4 - - [21/Nov/2021:12:08:01 +0000] "POST /ping//xxxx-xxx-xxxx-xxxx/start HTTP/1.1" 200 5767 "-" "runitor/v0.9.2 (+https://bdd.fi/x/runitor)"
1.2.3.4 - - [21/Nov/2021:12:08:01 +0000] "POST /ping//xxxx-xxx-xxxx-xxxx/0 HTTP/1.1" 200 1125 "-" "runitor/v0.9.2 (+https://bdd.fi/x/runitor)"
```
I can see both requests in my hc instance with a script that has no output.
My setup is Apache/2.4.29 + mod_wsgi 4.9

Awesome, thank you!
